File: android-java/app/src/main/java/com/oursky/skeleton/model/MyLoginSession.java
package com.oursky.skeleton.model;

import android.support.annotation.NonNull;

import com.oursky.skeleton.iface.SerializableToJson;

import org.json.JSONException;
import org.json.JSONObject;

@SuppressWarnings({"WeakerAccess", "unused"})
public class MyLoginSession implements SerializableToJson {
    public @NonNull String name;

    public MyLoginSession(@NonNull String name) {
        this.name = name;
    }
    public MyLoginSession(@NonNull MyLoginSession o) {
        name = o.name;
    }

    @Override
    public @NonNull JSONObject toJson() throws JSONException {
        return new JSONObject()
                .put("name", name);
    }
    public static @NonNull MyLoginSession from(@NonNull JSONObject json) throws JSONException, NullPointerException {
        return new MyLoginSession(
                json.getString("name")
        );
    }
}

File: android-kotlin/app/src/main/java/com/oursky/skeleton/model/MyLoginSession.kt
package com.oursky.skeleton.model

import org.json.JSONException
import org.json.JSONObject
import com.oursky.skeleton.iface.SerializableToJson

data class MyLoginSession(
        val id: Int = 0,
        val name: String = ""
) : SerializableToJson {
    @Throws(JSONException::class)
    override fun toJson(): JSONObject = JSONObject()
            .put("id", id)
            .put("name", name)

    companion object {
        @Throws(JSONException::class, NullPointerException::class)
        fun from(json: JSONObject): MyLoginSession {
            return MyLoginSession(
                    id = json.getInt("id"),
                    name = json.getString("name")
            )
        }
    }
}
